docs/modules/language-reference/pages/index.adoc

```diff
@@ -1990,10 +1990,10 @@ pigeon: ParentBird = new {
 ----
 
 [[methods]]
-== Methods
+== Functions and Methods
```

Comment:
Terminology-wise, we have two types of functions, methods, and lambdas (which we also called anonymous functions, and what `class Function` represents).

To eliminate confusion, I tend to avoid emphasizing the word "function", and instead prefer "method" and "lambda".

I think it's fine that we add a note in here that methods are functions, but I don't know if I want to make this a section header.